src/datasets/features/audio.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass, field
+from typing import Any, ClassVar, Optional
+
+import pyarrow as pa
+
+
+@dataclass
+class Audio:
+    """Audio Feature to extract audio data from an audio file.
+
+    Args:
+        coding_format (:obj:`str`, optional): Audio coding format. If `None`, this is inferred from file.
+    """
+
+    coding_format: str = None
+    id: Optional[str] = None
+    # Automatically constructed
+    dtype: ClassVar[str] = "dict"
+    pa_type: ClassVar[Any] = None
+    _type: str = field(default="Audio", init=False, repr=False)
+
+    def __post_init__(self):
+        self.coding_format = self.coding_format.upper() if self.coding_format else None
+
+    def __call__(self):
+        return pa.string()
+
+    def decode_example(self, value):
+        """Decode example audio file into audio data.
+
+        Args:
+            value: Audio file path.
+
+        Returns:
+            dict
+        """
+        import soundfile as sf
+
+        if self.coding_format and self.coding_format not in self._supported_coding_formats:
+            raise ValueError(
+                f"Audio coding format {self.coding_format} is not supported. Audio coding format must be "
+                f"one of: {self._supported_coding_formats}"
+            )
+
+        with open(value, "rb") as f:
```

Comment:
This is actually a bit problematic as it will always fail if the value is of type `.mp3` (which is Common Voice's format). To fix it, I think we should do the following:

```python
try:
    array, sample_rate = librosa.load(value, sr=self.sampling_rate, mono=self.mono)
except NoBackendError as e:
    raise ImportError(f"`librosa` is not able to load audio of format .{value.split(".")[-1]}). If you are trying to load MP3 audio files, please make sure to install `ffmpeg` as described here: https://github.com/librosa/librosa#audioread-and-mp3-support")
```

-> we should also add a couple of tests for `.mp3` IMO (one that fails gracefully and one that correctly loads MP3 with ffmpeg installed

Reply:
Also see: https://github.com/librosa/librosa/issues/945

Reply:
Would this be problematic for `streaming` @lhoestq ? I sadly don't think there is a way to open the file with and then pass the binary buffer to `librosa` for MP3

Reply:
For audio file support, I would like to again link to that excellent comment by `adefossez`: https://github.com/huggingface/transformers/pull/11337#discussion_r623779056

Shouldn't we have support for torchaudio too?

Reply:
For MP3 `torchaudio` is clearly the best choice, but it's a pretty heavy dependency with `torch`, but it's clearly the fastest. In my experiments, `torchaudio` gives a 10x speed up compared to `librosa` for MP3. Could we maybe add something like  "if format == mp3" -> check if torchaudio is installed - if not then throw error that `torchaudio` needs to be installed for `mp3`? 

Having played around with the feature a bit, `librosa + ffmpeg` is very very slow compared to `torchaudio` and I think the better choice is here to actually throw an error instead of falling back to `librosa + ffmpeg`.


@anton-l - do you know any fast decoding libraries apart `torchaudio`? What do you think?

Reply:
@patrickvonplaten if I understand, mp3 is not supported by `torchaudio` in Windows machines. Then, no support for mp3 for Windows?

Reply:
Yeah windows users will have to implement their own decoding then